modules/volume_access.cft.yaml

```diff
@@ -0,0 +1,473 @@
+            - "kms:ListResourceTags"
+            Resource: "*"
+          - Sid: "AllowKMSEncryptDecrypt"
+            Effect: "Allow"
+            Action:
+            - "kms:DescribeKey"
+            - "kms:Encrypt"
+            - "kms:Decrypt"
+            - "kms:ReEncrypt*"
+            - "kms:GenerateDataKey*"
+            - "kms:CreateGrant"
+            Resource: "*"
+            Condition:
+              StringLike:
+                "kms:ViaService": "ec2.*.amazonaws.com"
+          - Sid: "CreateTaggedSnapshotFromVolume"
+            Effect: "Allow"
+            Action:
+            - "ec2:CreateSnapshot"
+            Resource: "*"
+          - Sid: "CopySnapshots"
+            Effect: "Allow"
+            Action:
+            - "ec2:CopySnapshot"
+            Resource: "*"
+          - Sid: "SnapshotTags"
+            Effect: "Allow"
+            Action:
+            - "ec2:CreateTags"
+            Resource: "*"
+            Condition:
+              StringEquals:
+                "ec2:CreateAction": ["CreateSnapshot", "CopySnapshot"]
+                "aws:RequestTag/CreatedBy": "Sysdig"
+          - Sid: "ec2SnapshotShare"
+            Effect: "Allow"
+            Action:
+            - "ec2:ModifySnapshotAttribute"
+            Resource: "*"
+            Condition:
+              StringEqualsIgnoreCase:
+                "aws:ResourceTag/CreatedBy": "Sysdig"
+              StringEquals:
+                "ec2:Add/userId": !Ref ScanningAccountID
+          - Sid: "ec2SnapshotDelete"
+            Effect: "Allow"
+            Action:
+            - "ec2:DeleteSnapshot"
+            Resource: "*"
+            Condition:
+              StringEqualsIgnoreCase:
+                "aws:ResourceTag/CreatedBy": "Sysdig"    
+  AdministrationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub sysdig-secure-scanning-stackset-administration-${NameSuffix}
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: cloudformation.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Policies:
+      - PolicyName: !Sub sysdig-secure-scanning-stackset-execution-${NameSuffix}
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action:
+            - sts:AssumeRole
+            Resource:
+            - !Sub arn:aws:iam:::role/sysdig-secure-scanning-stackset-execution-${NameSuffix}
+  ExecutionRole:
+    Type: AWS::IAM::Role   
+    Properties:
+      RoleName: !Sub sysdig-secure-scanning-stackset-execution-${NameSuffix}
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Principal:
+            AWS:
+            - !GetAtt AdministrationRole.RoleId
+          Action:
+          - sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AdministratorAccess
```

Comment:
same here, too permissive? because mainly these roles require CloudFormation access and KMS access.

Reply:
updated